core/src/main/java/org/infinispan/remoting/transport/jgroups/JGroupsTransport.java

```diff
@@ -778,8 +766,9 @@ private boolean ignoreTimeout(ResponseFilter responseFilter) {
 
       Map<Address, Response> retval = new HashMap<>(CollectionFactory.computeCapacity(rsps.size()));
       boolean hasResponses = false;
+      Iterator<Map.Entry<Address, ReplicableCommand>> rpcCommandsIterator = rpcCommands.entrySet().iterator();
       for (Rsp<Response> rsp : rsps) {
```

Comment:
I made the change, but I just realized the loop is wrong. The entries in the map are not in the same order as the futures/rsps in the list.